openmc/material.py

```diff
@@ -331,6 +334,73 @@ def from_hdf5(cls, group: h5py.Group):
 
         return material
 
+    @classmethod
+    def from_ncrystal(cls, cfg, material_id=None, name=''):
+        """Create material from NCrystal configuration string. Density,
+        temperature, and material composition, and (ultimately) thermal neutron
+        scattering, will be automatically be provided by NCrystal based on this
+        string. The name and material_id parameters are simply passed on to the
+        Material constructor.
```

Comment:
Docstrings should start with a one-line summary followed by extra details:
```suggestion
        """Create material from NCrystal configuration string.
        
        Density, temperature, and material composition, and (ultimately) thermal
        neutron scattering will be automatically be provided by NCrystal based
        on this string. The name and material_id parameters are simply passed on
        to the Material constructor.
```